pkg/planner/core/collect_column_stats_usage.go

```diff
@@ -205,18 +205,17 @@ func (c *columnStatsUsageCollector) addHistNeededColumns(ds *DataSource) {
 		colIDSet.Insert(int(col.ID))
 		c.histNeededCols[tblColID] = true
 	}
-	for _, col := range ds.Columns {
+	for _, columns := range ds.tableInfo.Columns {
```

Comment:
```suggestion
	for _, column := range ds.tableInfo.Columns {
```

It seems that it is one column, not some columns.

Reply:
Done